### ISSUE TITLE
Frequent BSOD while working on wsl remote 

### ISSUE THREAD

**Issue by gulshan** [microsoft/vscode-remote-release#988]
Issue Type: <b>Bug</b>

I have experienced BSOD several times while wsl remote was running. But I could not find any specific sequence of tasks causing the BSOD. I am using the following extensions in WSL for Rust development-
- Rust (rls) 0.6.1
- Rust Test Explorer 0.7.0
- Test Explorer UI 2.13.0
- crates 0.4.3

Extension version: 0.38.10
VS Code version: Code 1.36.1 (2213894ea0415ee8c85c5eea0d0ff81ecc191529, 2019-07-08T22:59:35.033Z)
OS version: Windows_NT x64 10.0.18362
Remote OS version: Linux x64 4.4.0-18362-Microsoft

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz (4 x 2304)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.90GB (1.70GB free)|
|Process Argv|--folder-uri file:///d%3A/mstore-3.9.1/mstore-pro|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-18.04|
|OS|Linux x64 4.4.0-18362-Microsoft|
|CPUs|Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz (4 x 2304)|
|Memory (System)|7.90GB (1.70GB free)|
|VM|0%|
</details>

**Comment by nbdd0121** [comment]
I ran into the same issue 3 consecutive times today (all of them are `KERNEL_SECURITY_FAILURE`). I'll post details so hopefully it could be reproduced. I'm working on a Rust project with 90+ dependencies.

* For the first BSOD, I removed `target/`, which is quite large (16K+ files, 1G in size), and soon VSCode freezes, and bugcheck follows.
* After Windows reboots itself, I launched VSCode again, and RLS starts compilation. Another BSOD follows.
* After Windows reboots itself, I launched VSCode without problem. I then go to the project in bash and do `cargo build`, and Windows bugchecks for the third time.

I suspect that the BSODs are related to the WSL implementation of those inotify-related syscalls used by VSCode's watcher service, as Rust will create quite a large amount of time in a short period of time. After adding `"files.watcherExclude": { "**/target/**": true }"` into VSCode's setting, the problem disappears.
